wiki/Installing-Oppia-Android.md

```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
 
-4. Click on "Run" button to run robolectric test. (In my case "SplashActivityTest")
-   ![](https://github.com/oppia/oppia-android/assets/76530270/75a6b998-90c5-4f0a-8886-78f96970be90)
+```
+bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest
+```
 
-#### For tests that are in non-app modules, such as **domain** or **utility**::
+### Configure Robolectric Tests
+
+#### Step-by-Step guidance for setting up and running app module robolectric tests:
 
-1. In Android Studio, open the desired test file, e.g., `AnalyticsControllerTest`.
-2. In the test file, to the left of the class name, click on the orange and green arrow, and select **Run 'AnalyticsControllerTest'**.
-   - You will notice that the emulator is greyed out, but the run window will open to show the running tests:
-   ![](https://user-images.githubusercontent.com/59600948/272657015-158117e5-47d2-40fc-a38b-5dee6c347556.png)
+1. In Android Studio, open the test file that you wish to run. If sync completed successfully when the project was opened, there will be a green run arrow next to the class name, as well as next to each individual test name. Clicking on the arrow will run either the full class or a single test as selected.
 
-### Configure Emulator Tests
+2. Because we have set up the Bazel plugin, we can also run a test class using the run arrow at the top of the project window:
 
-**Espresso is slower for running tests, so we recommend using Robolectric.**
+   ![Screenshot 2025-02-26 at 22 16 32](https://github.com/user-attachments/assets/f3be5288-dc96-4079-bcb2-b514de81f899)
 
-1. In Android Studio, open the desired test file, e.g., `HomeActivityTest`.
-2. In the Android Studio toolbar, click on the `Available Devices` option. Select an emulator that has between API 28-30.
+   Edit configurations allows us to specify the run command:
+   ![Screenshot 2025-02-26 at 22 19 34](https://github.com/user-attachments/assets/462d07d2-4407-4898-bbc4-544509bd7486)
 
-   **Note**: If you don't have any available devices in this list, please follow [these instructions](#run-the-app-from-android-studio) to create one.
+   - Test **name**, is helpful for identifying the test target from the list on the left.
+   - **Target expression** requires the fully qualified path to the test file.
+   - **The Bazel command** for running tests is `test`
+   - **Bazel flags** are optional.
+   - Select **Apply** and then **Close** or **Ok**.
 
-3. In the test file, to the left of the class name, click on the orange and green arrow, and select **Run 'HomeActivityTest'**.
-   ![](https://user-images.githubusercontent.com/59600948/272657131-96e5354b-13a9-4709-969a-b9494a65c30f.png)
+#### For tests that are in non-app modules, such as **domain** or **utility**::
```

Comment:
Others:
- Can right click on the tab if the test is open and copy the BUILD target.
- For all tests (including app), you can run 'bazel query' on the relative source path of the file (also copyable via right click on the file tab when the test is open) and remove the '.kt' at the end of the returned value to get the test target.

Reply:
These are covered under `Set up and run tests` tips above.

Reply:
Transferred to teh to the user guide.